cli-plugins/apps/how-to-guides.hbs.md

```diff
@@ -0,0 +1,611 @@
+      status: "True"
+      type: ResourceSubmitted
+    ...
+    inputs:
+    - name: source-provider
+```
+
+## <a id='subpath-usage'> --sub-path
+
+`--sub-path` flag is used to define which path is used as the root path to apply the workload.
+
+- Git repository
+
+    ```bash
+    tanzu apps workload apply subpathtester --git-repo https://github.com/path-to-repo/my-repo --git-branch main --type web --sub-path my-subpath
+
+    🔎 Create workload:
+        1 + |---
+        2 + |apiVersion: carto.run/v1alpha1
+        3 + |kind: Workload
+        4 + |metadata:
+        5 + |  labels:
+        6 + |    apps.tanzu.vmware.com/workload-type: web
+        7 + |  name: subpathtester
+        8 + |  namespace: default
+        9 + |spec:
+       10 + |  source:
+       11 + |    git:
+       12 + |      ref:
+       13 + |        branch: main
+       14 + |      url: https://github.com/path-to-repo/my-repo
+       15 + |    subPath: my-subpath
+    ❓ Do you want to create this workload? [yN]:
+    ```
+
+- Local path
+  - In the directory of the project you want to create the workload from
+
+      ```bash
+      tanzu apps workload apply my-workload --local-path . -s gcr.io/my-registry/my-workload-image --sub-path subpath_folder
+      ❓ Publish source in "." to "gcr.io/my-registry/my-workload-image"? It may be visible to others who can pull images from that repository Yes
+      Publishing source in "." to "gcr.io/my-registry/my-workload-image"...
+      📥 Published source
+      
+      🔎 Create workload:
+            1 + |---
+            2 + |apiVersion: carto.run/v1alpha1
+            3 + |kind: Workload
+            4 + |metadata:
+            5 + |  name: myworkload
+            6 + |  namespace: default
+            7 + |spec:
+            8 + |  source:
+            9 + |    image: gcr.io/my-registry/my-workload-image:latest@sha256:f28c5fedd0e902800e6df9605ce5e20a8e835df9e87b1a0aa256666ea179fc3f
+           10 + |    subPath: subpath_folder
+      ❓ Do you want to create this workload? [yN]:
+
+      ```
+
+## <a id='tanzuignore-file-usage'> .tanzuignore file
+
+As more systems around us adopt the "as code" approaches, application developers will increasingly have files in their projects that have nothing to do with actually running code (those files don't end up in the running container).
+
+When creating a workload from local source code, these unused files can be added to the `.tanzuignore` file so there won't be an unnecesary consumption of resources and also, the workload deployment won't be triggered when one of those files changes. The file must contain itself as well for it to be ignored.
+
+The `.tanzuignore` file must be located at the root of the project so Apps CLI plug-in can read it.
+It allows to list folders (these must not end with the system separator, e.g. `/` or `\`), files and have comments which start with `#`. If the file contains files or directories that are not in the source code, they are ignored.
```

Comment:
Folders/directories are supported (these must not end with the system separator, e.g. `/` or `\`)

Individual files can be listed. 

And comments (which start with `#`) can be included . 

If the `.tanzuignore` file contains files or directories that are not found in the source code, they will be ignored.